File: android-app/src/main/java/com/jaa/likeastarappmpp/units/FilmTableDataFactory.kt
package com.jaa.likeastarappmpp.units

import com.jaa.library.feature.filmList.model.FilmRowData
import com.jaa.library.feature.filmList.presentation.FilmTableDataFactoryInterface
import com.jaa.likeastarappmpp.FilmListRow
import dev.icerock.moko.units.TableUnitItem

class FilmTableDataFactory : FilmTableDataFactoryInterface {

    override fun createFilmRow(
        id: Long,
        film: FilmRowData,
        listener: FilmTableDataFactoryInterface.ListRowTappedListener
    ): TableUnitItem {
        return FilmListRow().apply {
            itemId = id
            this.film = film
            this.listener = listener
        }
    }

}

File: ios-app/src/view/FilmTableDataFactory.swift
import UIKit
import MultiPlatformLibrary
import MultiPlatformLibraryUnits

import UIKit

class FilmTableDataFactory: FilmTableDataFactoryInterface {

    func createFilmRow(id: Int64, film: FilmRowData, listener: FilmTableDataFactoryInterfaceListRowTappedListener) -> TableUnitItem {
        return UITableViewCellUnit<FilmTableViewCell>(
            data: FilmTableViewCell.CellModel(
                id: id,
                filmData: film,
                listener: listener
            ),
            itemId: id,
            configurator: nil
        )
    }
    
}
